feat: Add bedrock async support

miri-bar opened the pull request:
In this PR:
* Replaced the usage of boto3 with a custom aws http client (shouldn't affect the way the user uses the Bedrock client).
* Async support for Bedrock client.

Josephasafg commented on ai21/clients/bedrock/resources/bedrock_completion.py:
```diff
@@ -46,6 +46,47 @@ def create(
             raise ValueError("model_id should be provided in either the constructor or the 'create' method call")
 
-        raw_response = self._invoke(model_id=model_id, body=body)
+        raw_response = self._post(body=body, model_id=model_id)
```
Is there a reason to keep this class? Can we delete this and use `ai21/clients/studio/resources/chat/chat_completions.py` instead?

Josephasafg commented:
@miri-bar We've added support for Jamba in Bedrock, so please make sure you update it as well :)

Apologies for the conflicts 🙏🏼

Josephasafg commented on ai21/clients/aws_http_client/aws_http_client.py:
```diff
@@ -0,0 +1,240 @@
+DEFAULT_AWS_REGION = "us-east-1"
+
+
+class BaseAWSHttpClient(ABC):
```
I was actually thinking about this. This class might not be necessary at all and you could use the regular http client that we have. let's talk about this

miri-bar commented on ai21/errors.py:
```diff
@@ -41,6 +41,11 @@ def __init__(self, details: Optional[str] = None):
 
+class ModelStreamError(AI21APIError):
+    def __init__(self, details: Optional[str] = None):
+        super().__init__(424, details)
```
https://docs.aws.amazon.com/bedrock/latest/APIReference/API_runtime_ResponseStream.html
Per their docs

miri-bar commented on ai21/clients/bedrock/ai21_bedrock_client.py:
```diff
@@ -1,24 +1,62 @@
         model_id: Optional[str] = None,
-        session: Optional[boto3.Session] = None,
         region: Optional[str] = None,
+        headers: Optional[Dict[str, Any]] = None,
```
Irrelevant, since we removed AWSHttpClient :)

miri-bar commented on ai21/clients/bedrock/bedrock_session.py:
```diff
@@ -1,63 +0,0 @@
-
-    def _handle_client_error(self, client_exception: ClientError) -> None:
-        error_response = client_exception.response
-        error_message = error_response.get("Error", {}).get("Message", "")
-        status_code = error_response.get("ResponseMetadata", {}).get("HTTPStatusCode", None)
```
I'm handling the Bedrock-specific errors in the bedrock client

miri-bar commented:
@Josephasafg thanks for the review!
Fixed the code per your suggestions and replied to your comments.
I refactored the code and am no longer using the aws_http_client, but the regular http client, so can you please review it again?
I've added a test for the aws auth class, are there other tests you think we should add for this class?
